Code/Editor/Plugins/ProjectSettingsTool/ProjectSettingsContainer.h

```diff
@@ -26,89 +27,106 @@ namespace ProjectSettingsTool
 
     struct SettingsError
     {
-        SettingsError(const AZStd::string& error, const AZStd::string& reason);
+        SettingsError(const AZStd::string& error, const AZStd::string& reason, bool shouldAbort = false);
 
         // The error that occurred
         AZStd::string m_error;
+
         // The reason the error occurred
         AZStd::string m_reason;
+
+        bool m_shouldAbort = false;
     };
 
-    // Loads, saves, and provides access to all of the project settings files
+    // Loads, saves, and provides access to all of the project settings files of all platforms.
+    // It handles base settings and platform settings (android, ios) separately.
+    // For base settings it uses json document and for platform settings it uses different documents
+    // depending on the type, it supports json and plist formats.
     class ProjectSettingsContainer
     {
     public:
-        typedef AZStd::pair<PlatformId, AZStd::string> PlatformAndPath;
-        typedef AZStd::vector<PlatformAndPath> PlistInitVector;
+        using PlatformAndPath = AZStd::pair<PlatformId, AZStd::string>;
+        using PlatformResources = AZStd::vector<PlatformAndPath>;
 
         template<class DocType>
-        struct PlatformSettings
+        struct Settings
         {
             // File path to document
             AZStd::string m_path;
+
             // Raw string loaded from file
             AZStd::string m_rawData;
+
             // The document itself
             AZStd::unique_ptr<DocType> m_document;
 
         };
 
-        using JsonSettings = PlatformSettings<rapidjson::Document>;
-        using PlistSettings = PlatformSettings<AZ::rapidxml::xml_document<char>>;
+        using JsonSettings = Settings<rapidjson::Document>;
+        using PlistSettings = Settings<AZ::rapidxml::xml_document<char>>;
```

Comment:
Would it be okay to bring the `XmlDocument` alias over from the cpp file?  If it's defined inside the class then it should be scoped to `ProjectSettingsContainer` and not pollute the global namespace.

Reply:
`XmlDocument` alias is defined in `PlistDictionary.cpp` under `ProjectSettingsTool` namespace and used by `PlistDictionary` class. I'll move it to `PlistDictionary.h` so `ProjectSettingsContainer` can use it for its `PlistSetings` alias.

Thanks!

Reply:
Ohh and it was duplicated in `ProjectSettignsContainer.cpp`, not sure why it was done that way.

Reply:
I did a clean up pass on `XmlDocument` and `XmlNode` aliases. 

More files will show up in this PR and there are other things that can be cleaned up, but I'm going to have to push back on any further cleaning on this plugin, I need to get this in and move on with android and VR tasks :)